File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int maxNum = 0;
        
        for (int i = 1; i <= Math.max(n, m); i++) {
            if (n % i == 0 && m % i == 0) maxNum = Math.max(maxNum, i);
        } 
        int minNum = (n / maxNum) * (m / maxNum) * maxNum;
        
        return new int[] {maxNum, minNum};
        
        
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
// 8, 12 [1,2,4,8] [1,2,3,4,6,12]

const getMax = (n, m) => {
    let result = 1;
    for (let i = n; i > 0; i--) {
        if (n % i === 0) {
            if (m % i === 0) {
                result = i;
                break;
            }
        }
    }
    
    return result;
}

const getMin = (n, m, max) => {
    return (n / max) * (m / max) * max;
}

function solution(n, m) {
    const a = getMax(n,m);
    return [a, getMin(n,m,a)];
}